packages/client/TODO.md

```diff
@@ -0,0 +1,160 @@
+- [H] client.profile.isValidHandle(handle)
+
+- [AM] client.profile.createSetProfileMetadataTypedData(request)
+- [AM] client.profile.createSetProfileMetadataViaDispatcher(request)
+
+- [AM] client.profile.createSetProfileImageURITypedData(request)
+- [AM] client.profile.createSetProfileImageURIViaDispatcher(request)
+
+- [AM] client.profile.createBurnProfileTypedData(request)
+- [AM] client.profile.createSetDefaultProfileTypedData(request)
+- [AM] client.profile.createSetDispatcherTypedData(request)
+
+- [AM] client.profile.addInterests(request)
+- [AM] client.profile.removeInterests(request)
+
+## Publication(s)
+
+### Query publications
+
+- [Q] client.publication.fetch(request) or client.publications.fetchOne()
+- [PQ] client.publication.fetchAll(request)
+- [Q] client.publication.validateMetadata(request)
+- [PQ] client.publication.whoCollected({ publicationId })
+
+- [PQ] client.publication(s).forSale({ profileId })
+- [Q] client.publication.metadataStatus(request)
+
+### Create or edit publications
+
+- [AM] client.publication.createPostTypedData(request)
+- [AM] client.publication.createPostViaDispatcher(request)
+
+- [AM] client.publication.createCommentTypedData(request)
+- [AM] client.publication.createCommentViaDispatcher(request)
+
+- [AM] client.publication.createMirrorTypedData(request)
+- [AM] client.publication.createMirrorViaDispatcher(request)
+
+- [AM] client.publication.hide({ publicationId })
+- [M] client.publication.createAttachMediaData(request)
+
+- gated publications TBD
+
+## ProtocolStats
+
+- [Q] client.protocolStats.fetch(request)
+
+## UserSigNonces
+
+- [AQ] client.nonces.fetch()
+
+## Proxy actions
+
+- [AM] client.proxyActions.freeFollow({ profileId })
+- [AM] client.proxyActions.freeCollect({ publicationId })
+- [AQ] client.proxyActions.checkStatus({ proxyActionId })
+
+## Reaction(s)
+
+- [x] [AM] client.reaction.add(request)
+- [x] [AM] client.reaction.remove(request)
+- [PQ] client.reaction(s).toPublication({ publicationId })
+
+## Reporting
+
+- [AM] client.report.publication({ publicationId, reason, additionalComments })
+
+## Revenue
+
+- [PQ] client.revenue.profilePublications({ profileId, limit })
+- [Q] client.revenue.profileFollow({ profileId })
+- [Q] client.revenue.publication({ publicationId })
+
+## Search
+
+- [PQ] client.search.profiles({ query, limit })
+- [PQ] client.search.publications({ query, limit })
+
+## Feed
+
+- [PQ] client.feed.fetch({ profileId })
+
+## Other considerations
+
+- paginated response helper
+- more helpers like `isValidHandle`
+- how to split a group that has both authenticated and public requests?
+- authentication required error thrown on a group or on a method call?
+- allow for custom gql queries or compose query from subqueries (like profile base, profile interests, profile onChainIdentity etc.) (is that possible?)
+
+```ts
+client.profile.fetch(request, gql: {
+    ...baseProfileQuery,
+    ...profileInterestsQuery,
+    ...profileOnChainIdentityQuery
+})
+```
```

Comment:
Wow, thank you for doing this analysis. It's a long list!

Reply:
Enormous effort 🥇